### PR TITLE
Workaround for legacy message without metadata during request/reply

### PR DESCRIPTION
**Pull request by ericclappier-eaton**

### CODE REVIEW

File: src/fty_common_messagebus_malamute.cc [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 namespace messagebus {
 
-    static Message _fromZmsg(zmsg_t *msg) {
+    static Message _fromZmsg(zmsg_t *msg, std::string from) {
```
Comment (boricj) [inline src/fty_common_messagebus_malamute.cc]: `from` parameter should be const reference.

**Review by NavarroPierre — APPROVED** [review]

File: src/fty_common_messagebus_malamute.cc [hunk 2]
```diff
@@ -329,19 +332,25 @@ namespace messagebus {
...
-                    std::unique_lock<std::mutex> lock(m_cv_mtx);
-                    m_syncResponse = msg;
-                    m_cv.notify_one();
-                    m_syncUuid = "";
-                    syncResponse = true;
-                }
+                uuid_read = it->second;
+            }
+            else {
+                // Get uuid in fist part of message (for legacy message without meta data)
```
Comment (perrettecl) [inline src/fty_common_messagebus_malamute.cc]: We shouldn't implement any protocol here. The client will have to manage the protocol his self. UUID is part of the protocol.

I would prefer that if the request is RAW format, we transfer the first reply in case of sync.

File: src/fty_common_messagebus_malamute.cc [hunk 3]
```diff
@@ -414,6 +423,21 @@ namespace messagebus {
...
             }
         }
+        // then try with address of message (= <topic name>)
+        else {
+            const char *address = mlm_client_address(m_client);
+            if (address) {
+                iterator = m_subscriptions.find (address);
+                if (iterator != m_subscriptions.end ()) {
+                    //iterator->second(msg);
+                    std::thread (iterator->second, msg).detach();
```
Comment (perrettecl) [inline src/fty_common_messagebus_malamute.cc]: we should not detach thread in the implementation (cf recent merge in the master). Client will do it if he needs / wants

File: src/fty_common_messagebus_malamute.cc [hunk 4]
```diff
@@ -55,6 +55,9 @@ namespace messagebus {
                 }
             }
             else {
+                // Workaround for legacy message without metadata:
+                // For request/reply, need at least "from" meta data for reply management
+                message.metaData().emplace(Message::FROM, from);
```
Comment (perrettecl) [inline src/fty_common_messagebus_malamute.cc:60]: If we want to support "RAW" data by this way, we should also set "RAW" flag in the meta data

**Review by perrettecl — CHANGES_REQUESTED** [review]

File: src/fty_common_messagebus_malamute.cc (hunk 4 above)
Comment (ericclappier-eaton) [reply src/fty_common_messagebus_malamute.cc:60]: You are right, it is better to add "RAW" flag in meta data. Take into account.

File: src/fty_common_messagebus_malamute.cc (hunk 3 above)
Comment (ericclappier-eaton) [reply src/fty_common_messagebus_malamute.cc]: Take into account

**Review by boricj — APPROVED** [review]

File: src/fty_common_messagebus_malamute.cc (hunk 2 above)
Comment (ericclappier-eaton) [reply src/fty_common_messagebus_malamute.cc]: Take into account
